### PR TITLE
feat: add Uniswap V3 Create Pool Action

### PR DESCRIPTION
**Pull request by majkic99**

### CODE REVIEW

File: contracts/actions/uniswap/v3/UniCreatePoolV3.sol [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    /// @param tickUpper The higher end of the tick range for the position
+    /// @param amount0Desired The desired amount of token0 that should be supplied
+    /// @param amount1Desired The desired amount of token1 that should be supplied
+    /// @param amount0Min The minimum amount of token0 that should be supplied,
+    /// @param amount1Min The minimum amount of token1 that should be supplied,
+    /// @param recipient address which will receive the NFT
+    /// @param deadline The time by which the transaction must be included to effect the change
+    /// @param from account to take amounts from
+    /// @param sqrtPriceX96 The initial square root price of the pool as a Q64.96 value
+    /// @return tokenId The ID of the token that represents the position
```
Comment (nklipa13) [inline contracts/actions/uniswap/v3/UniCreatePoolV3.sol:30]: Struct doesn't have a return value

File: contracts/actions/uniswap/v3/UniCreatePoolV3.sol [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    }
+
+    /// @inheritdoc ActionBase
+    function actionType() public pure virtual override returns (uint8) {
+        return uint8(ActionType.STANDARD_ACTION);
+    }
+
+    //////////////////////////// ACTION LOGIC ////////////////////////////
+
+    function _createPool(Params memory _inputData) internal returns (address pool) {
```
Comment (nklipa13) [inline contracts/actions/uniswap/v3/UniCreatePoolV3.sol:91]: Why are we returning pool address if we are never using that param?

File: contracts/actions/uniswap/v3/UniCreatePoolV3.sol [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        logger.Log(
+            address(this),
+            msg.sender,
+            "UniCreatePoolV3",
+            abi.encode(_inputData, tokenId, liquidity, amount0, amount1)
+        );
+    }
+
+    /// @dev mints new NFT that represents a position with selected parameters
+    /// @return tokenId of new NFT, how much liquidity it now has and token amounts
```
Comment (nklipa13) [inline contracts/actions/uniswap/v3/UniCreatePoolV3.sol:136]: Describe what are token amounts, is it the leftovers or is it how much is actually put in the pool

File: test/uniswap/v3/uni-create-pool.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    };
+
+    before(async () => {
+        await redeploy('UniCreatePoolV3');
+        senderAcc = (await hre.ethers.getSigners())[0];
+        proxy = await getProxy(senderAcc.address);
+        positionManager = await hre.ethers.getContractAt('IUniswapV3NonfungiblePositionManager', UNIV3POSITIONMANAGER_ADDR);
+    });
+
+    it('create a pool that does not exist yet', async () => {
```
Comment (nklipa13) [inline test/uniswap/v3/uni-create-pool.js:34]: Write a test where it should return some leftovers and check if thats handled properly

**Review by nklipa13 — CHANGES_REQUESTED** [review]

File: test/uniswap/v3/uni-create-pool.js (hunk 4 above)
Comment (majkic99) [reply test/uniswap/v3/uni-create-pool.js:34]: test already checks if proxy balance before is the same as proxy balance after => meaning that any tokens that weren't sent to uniswap pool are returned to EOA

**Review by nklipa13 — APPROVED** [review]

**Pull request merged**